geopandas/tests/test_merge.py

```diff
@@ -97,3 +98,23 @@ def test_concat_axis1_multiple_geodataframes(self):
         res3 = pd.concat([df2.set_crs("epsg:4326"), self.gdf], axis=1)
         # check metadata comes from first df
         self._check_metadata(res3, geometry_column_name="geom", crs="epsg:4326")
+
+    @pytest.mark.xfail(
+        not compat.PANDAS_GE_11,
+        reason="pandas <=1.0 hard codes concat([GeoSeries, GeoSeries]) -> "
+        "DataFrame or Union[DataFrame, SparseDataFrame] in 0.25",
+    )
+    def test_concat_axis1_geoseries(self):
+        gseries2 = GeoSeries([Point(i, i) for i in range(3, 6)], crs="epsg:4326")
+        result = pd.concat([gseries2, self.gseries], axis=1)
+        # result columns are [0, 1], not an explicit way to rename them.
+        # leave geometry column as None and user to sort out.
+        # This could happen in a less contrived way via DataFrame.convert_dtypes
+        assert type(result) is GeoDataFrame
+        assert result._geometry_column_name is None
```

Comment:
I am wondering is it's actually useful that the result is a GeoDataFrame, if we don't set any of the columns as the geometry column. The user will always need to call `set_geometry` to fix that up, but you can also do that on a DataFrame (and the fact that it's already a GeoDataFrame might then be confusing)

Reply:
Or, seeing the test below, is this in this case only because the GeoSeries inputs have no name?

Reply:
This was meant to be mocking the cases inside concat (see example below for what I mean by this), but concat actually has names. So it's a bad test in review, and I don't think the behaviour within is what should happen in this case, because it's not consistent with the named case. I think for the purpose of concat, GeoSeries should mimic GeoDataFrame with a single column that is set as the geometry column. Based on that rule, the left hand entry should be set as the geometry column (with the default column name of "0" - perhaps there's an argument that should be translated to "geometry" as well, but check for this would probably introduce more edge cases), and the expected behaviour of this test should change. 

To do that with this solution, and extra edge case needs to go into the finalize of concat (not to copy the name from the GeoSeries if the geoseries name is none). But I wonder (in conjunction with the above if there is a better solution by having expanddim construct a pd.DataFrame first then alter this to a GeoDataFrame. 

Alternatively there's perhaps an argument that concat could be nicer on the pandas side, but that won't help with older versions of pandas.

The example - which happens from eg convert_dtypes (which to be honest is probably not all that relevant, but digging through it has helped me work out what I've said above):
```python
import pandas as pd
import geopandas as gpd
from shapely.geometry import Point

geoms = [Point(0, 0), Point(1, 1)]
# name is required in geoseries, suggests we need a better solution
srs = gpd.GeoSeries(geoms, crs=4326, name='geometry')
srs2 = pd.Series([0,1], name='col1')

df = pd.concat([srs, srs2], axis=1)
print(df)
print(type(df))
print(df.geometry.name)
##################
                  geometry  col1
0  POINT (0.00000 0.00000)     0
1  POINT (1.00000 1.00000)     1
<class 'geopandas.geodataframe.GeoDataFrame'>
geometry
```
Inside `pd.concat` (`pandas/core/reshape/concat.py::get_result(self)`) [https://github.com/pandas-dev/pandas/blob/8814de9fbc7d53c6db201eab717fd3d6b607654b/pandas/core/reshape/concat.py#L530]
`self_is_series` is True, `self.bm_axis == 0` is False, so we get
```python
data = dict(zip(range(len(self.objs)), self.objs))
# GH28330 Preserves subclassed objects through concat
cons = sample._constructor_expanddim

index, columns = self.new_axes
df = cons(data, index=index)
df.columns = columns
return df.__finalize__(self, method="concat")
```
With some debugging
```python
data.keys()
Out[2]: dict_keys([0, 1])
data.values()
Out[3]: 
dict_values([0    POINT (0.00000 0.00000)
1    POINT (1.00000 1.00000)
Name: geometry, dtype: geometry, 0    0
```
So the arguments to `_constructor_expanddim` strip out the column names. 

```python
df = cons(data, index=index)
df
Out[2]: 
                         0  1
0  POINT (0.00000 0.00000)  0
1  POINT (1.00000 1.00000)  1
df.geometry.name
Out[3]: 0
```
After the column name setting and finalize the column and geometry names get updated again:
```python
df
Out[4]: 
                  geometry  col1
0  POINT (0.00000 0.00000)     0
1  POINT (1.00000 1.00000)     1
df.geometry.name
Out[5]: 'geometry'
```